app/controllers/claims/support/schools/claims_controller.rb

```diff
@@ -1,18 +1,27 @@
 class Claims::Support::Schools::ClaimsController < Claims::Support::ApplicationController
   include Claims::BelongsToSchool
 
-  before_action :set_claim, only: %i[check draft show edit update]
+  before_action :set_claim, only: %i[check draft show edit update remove destroy]
   before_action :authorize_claim
   helper_method :claim_provider_form
 
   def index
-    @pagy, @claims = pagy(@school.claims.not_internal.order_created_at_desc)
+    @pagy, @claims = pagy(@school.claims.not_internal_or_discarded.order_created_at_desc)
   end
 
   def new; end
 
   def show; end
 
+  def remove; end
+
+  def destroy
+    @claim.destroy! if @claim.draft?
+    discard_submitted_claim if @claim.submitted?
```

Comment:
Is there a case where the claim would not be either `draft` or `submitted`?
```suggestion
    @claim.submitted? ? @claim.discard : @claim.destroy!
```

Reply:
I'm wondering if it'd be worth using soft-deletion everywhere for the Claim model?

It feels a bit unusual to implement soft-delete only for 'submitted' claims, and instead to permanently delete 'draft' claims. I could imagine that being confusing behaviour – for example, if somebody accidentally deletes a 'submitted' claim, we'd be able to help them retrieve it by un-discarding it. But the same wouldn't be true if the claim happened to have been in a 'draft' state, since it would have been permanently deleted.

This is more of a data modelling decision rather than an implementation detail, so I'm wondering if @Nitemaeric has an opinion? It's the sort of thing that's worth getting right upfront.

And as @JamieCleare2525 said, we implemented the [discard](https://github.com/jhawthorn/discard) gem back in PR #222, so it'd be good to follow that approach as a consistent way to soft-delete records within the app.

Reply:
This is a Product requirement we've received, https://trello.com/c/JUhoXxIJ/259-allow-support-users-to-remove-claims#comment-65fbedab17e1be89451976e0. We've decided that "archived" is equal to "soft-deleted" and opted towards using the `discard` gem to manage that.

The support interface will have a way to view "archived" records, so we want to make the distinction between hard-deleting drafts and soft-deleting (archiving) submitted claims.

We _may_ be incorrectly equating "archived" and "soft-deleted". This is open for debate.